src/crypto/local.js

```diff
@@ -313,22 +313,17 @@ function recomputeComposites(chkM, chkZ, pointG, pointH) {
     let seed = getSeedPRNG(chkM, chkZ, pointG, pointH);
     let shake = createShake256();
     shake.update(seed, "hex");
-    let cM;
-    let cZ;
+    let cM = new sjcl.ecc.pointJac(CURVE); // can only add points in jacobian representation
+    let cZ = new sjcl.ecc.pointJac(CURVE);
     for (let i=0; i<chkM.length; i++) {
         let ci = getShakeScalar(shake);
         let cMi = _scalarMult(ci, chkM[i].point);
         let cZi = _scalarMult(ci, chkZ[i]);
-        if (cM === undefined || cZ === undefined) {
-            cM = cMi;
-            cZ = cZi;
-        } else {
-            cM = cM.toJac().add(cMi).toAffine();
-            cZ = cZ.toJac().add(cZi).toAffine();
-        }
+        cM = cM.add(cMi);
+        cZ = cZ.add(cZi);
     }
 
-    return {M: cM, Z: cZ};
+    return {M: cM.toAffine(), Z: cZ.toAffine()};
```

Comment:
You can delay the conversion to affine, because the caller function uses these point to perform another scalar multiplication.

Reply:
Another related optimization that works is replacing two consecutive scalar mults by calling the method `mult2` (which calculates `aP+bQ`) provided by sjcl.


For example in lines (275-276) and (280-281)

Reply:
Good point, I'll see if I can add this in a future change